compiler/parser/RdrHsSyn.hs

```diff
@@ -4473,6 +4474,16 @@ warnStarBndr span = addWarning Opt_WarnStarBinder span msg
            <+> text "modules with StarIsType,"
         $$ text "    including the definition module, you must qualify it."
 
+warnRetroactiveInterfaceInstance :: SrcSpan -> P ()
+warnRetroactiveInterfaceInstance span =
+  addWarning Opt_WarnRetroactiveInterfaceInstances span msg
+  where
+    msg =  text "Found interface instance in the body of an interface" <+>
+           parens (text "a" <+> doubleQuotes (text "retroactive interface instance")) <> dot
+        $$ text "This feature is deprecated, it will be removed in a future version of Daml."
+        $$ text "Instead, consider defining the interface instance in the body of the template." <+>
+           text "If the template is defined in another package, this will require upgrading that package."
```

Comment:
Is the use of the word "upgrading" here referring to upgrades as a feature, or just to the general concept of updating a package?

If the former case is true, maybe we can link the docs?

If the latter case is true, it might be best to use "updates" to avoid confusion with the upgrades feature.

Reply:
I see that it's the former case - up to you if you want to link the docs.

Reply:
According to Milestone 1 on https://docs.google.com/document/d/10q3LYdW4FbtORhPtzOpJDoHSe64LEyNrulqdLFF4AFc/edit , docs are still in-progress, Bas is working on them so he'd be best to ping.

In the meantime to not block this, probably best to leave a TODO?